src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoCurrencyNegativePattern.cs

```diff
@@ -2,95 +2,51 @@
 
+using System.Collections.Generic;
 using System.Linq;
-using System.Runtime.InteropServices;
 using Xunit;
 
 namespace System.Globalization.Tests
 {
     public class NumberFormatInfoCurrencyNegativePattern
     {
-        // TestDefault: Verify default value of property CurrencyNegativePattern
-        [Fact]
-        public void TestDefault()
-        {
-            NumberFormatInfo nfi = new NumberFormatInfo();
-            int expected = nfi.CurrencyNegativePattern;
-            Assert.Equal(0, expected);
-        }
-
-        // TestSetValue: Verify set value of property CurrencyNegativePattern
-        [Fact]
-        public void TestSetValue()
-        {
-            NumberFormatInfo nfi = new NumberFormatInfo();
-            for (int i = 0; i <= 15; i++)
-            {
-                nfi.CurrencyNegativePattern = i;
-                Assert.Equal(i, nfi.CurrencyNegativePattern);
-            }
-        }
-
-        // TestArgumentOutOfRangeException: ArgumentOutOfRangeException is thrown
-        [Fact]
-        public void TestArgumentOutOfRangeException()
-        {
-            VerificationHelper<ArgumentOutOfRangeException>(-1);
-            VerificationHelper<ArgumentOutOfRangeException>(16);
-        }
-
-        // TestInvalidOperationException: InvalidOperationException is thrown
-        [Fact]
-        public void TestInvalidOperationException()
+        public static IEnumerable<object[]> CurrencyNegativePattern_TestData()
         {
-            NumberFormatInfo nfi = new NumberFormatInfo();
-            NumberFormatInfo nfiReadOnly = NumberFormatInfo.ReadOnly(nfi);
-            Assert.Throws<InvalidOperationException>(() =>
-            {
-                nfiReadOnly.CurrencyNegativePattern = 1;
-            });
+            yield return new object[] { NumberFormatInfo.InvariantInfo, new int[] { 0 } };
+            yield return new object[] { new CultureInfo("bg-BG").NumberFormat, new int[] { 8 } };
+            yield return new object[] { new CultureInfo("en-US").NumberFormat, NumberFormatInfoData.GetCurrencyNegativePatterns("en-US") };
+            yield return new object[] { new CultureInfo("en-CA").NumberFormat, NumberFormatInfoData.GetCurrencyNegativePatterns("en-CA") };
+            yield return new object[] { new CultureInfo("fa-IR").NumberFormat, NumberFormatInfoData.GetCurrencyNegativePatterns("fa-IR") };
+            yield return new object[] { new CultureInfo("fr-CD").NumberFormat, NumberFormatInfoData.GetCurrencyNegativePatterns("fr-CD") };
+            yield return new object[] { new CultureInfo("as").NumberFormat, NumberFormatInfoData.GetCurrencyNegativePatterns("as") };
+            yield return new object[] { new CultureInfo("es-BO").NumberFormat, NumberFormatInfoData.GetCurrencyNegativePatterns("es-BO") };
+            yield return new object[] { new CultureInfo("fr-CA").NumberFormat, NumberFormatInfoData.GetCurrencyNegativePatterns("fr-CA") };
```

Comment:
Why is this transformation better than the original?  Seems like lots of duplicated code that the original didn't have.